### ISSUE TITLE
Confusing VisibleDeprecationWarning when multiplying a numpy.float64 with None

### ISSUE THREAD

**Issue by gomyhr** [numpy/numpy#7891]
When multiplying a numpy.float64 by None (by mistake), I get a warning which seems out of place:
VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

f = np.float64(0.63)
x = f \* None

triggers the warning, whereas

f = np.float(0.63)
x = f \* None

does not (since it uses native float type).

Using numpy 1.11.1 from anaconda on 64-bit linux.

**Comment by charris** [comment]
Master raises a `TypeError` for both cases. Note that warnings only get issued once unless you do something like
```
In [2]: import warnings

In [3]: warnings.simplefilter("always")
```
before anything else.

**Comment by gomyhr** [comment]
Yes, the TypeError is the expected behaviour. It is the warning before it that is confusing.

And although I was confused at first when the warning was issued only once (when trying to trigger it), the second example does not issue a warning in any case.

**Comment by charris** [comment]
Hmm, it only appears only with `*`, not with `+` or `/`. Looks like it is in the `gentype_multiply` function in `scalartypes.c.src` where an attempt is made to convert an argument to `int`. Because the warning will go away in the next release it probably isn't worth trying to fix for 1.11.x, although the `gentype_multiply` function itself could maybe use an audit.

**Comment by mattip** [comment]
Closing, 1.11 is not going to get a bug-fix release. Please reopen if I misunderstood.
